transpiler/src/target-ast/node-transformers/controllers/index.ts

```diff
@@ -0,0 +1,44 @@
+import { ControllerNode } from '../../../ast/core/intermediate-ast/nodes/controllers/ControllerNode.js';
+import { GraphQLControllerNode } from '../../../ast/core/intermediate-ast/nodes/controllers/graphql/GraphQLControllerNode.js';
+import { RESTControllerNode } from '../../../ast/core/intermediate-ast/nodes/controllers/restController/RESTControllerNode.js';
+import { NodeModelToTargetASTTransformer } from '../index.js';
+
+class BaseControllerNodeTSTransformer<
+  T extends ControllerNode,
+> extends NodeModelToTargetASTTransformer<T> {
+  run(): void {
+    this.transformAwait();
+    this.transformDotValue();
+  }
+
+  private transformAwait(): void {
+    const executeStatement = this.tree.getUseCaseExecuteStatementOf(this.node);
+    if (!executeStatement) {
+      return;
+    }
+
+    const expression = executeStatement.getExpression();
+    if (!expression.isMethodCallExpression()) {
+      return;
+    }
+    expression.prependAwaitToThisMethod();
+    return this.tree.buildValueRecursiveBottomUp(this.node);
+  }
+
+  private transformDotValue(): void {
+    const executeStatement = this.tree.getUseCaseExecuteStatementOf(this.node);
+    if (!executeStatement) {
+      return;
+    }
+
+    const identifierNode = executeStatement.getIdentifier();
+    const identifierValue = identifierNode.getIdentifierName();
+    const newValue = `${identifierValue}.value`;
```

Comment:
newValue could be renamed to something indicating more what this is and the right part maybe a private method if we are going to use it more. If not that's ok.